processor/processor.go

```diff
@@ -531,6 +535,9 @@ func (proc *Handle) activePartitions(ctx context.Context) []string {
 }
 
 func (proc *Handle) Shutdown() {
+	if proc.dedup != nil {
+		proc.dedup.Close()
+	}
```

Comment:
Let's move this as last operation, since dedup is a dependency for processor, meaning that first we should stop processor's loops and then its dependencies.

Reply:
Done